library/ecp_curves.c

```diff
@@ -5213,40 +5214,29 @@ static int ecp_mod_p521( mbedtls_mpi *N )
 
 /*
  * Fast quasi-reduction modulo p255 = 2^255 - 19
- * Write N as A0 + 2^255 A1, return A0 + 19 * A1
+ * Write N as A0 + 2^256 A1, return A0 + 38 * A1
  */
 static int ecp_mod_p255( mbedtls_mpi *N )
 {
-    int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
-    size_t i;
-    mbedtls_mpi M;
-    mbedtls_mpi_uint Mp[P255_WIDTH + 2];
+    mbedtls_mpi_uint Mp[P255_WIDTH];
 
-    if( N->n < P255_WIDTH )
+    /* Helper references for top part of N */
+    mbedtls_mpi_uint * const NT_p = N->p + P255_WIDTH;
+    size_t const NT_n = N->n - P255_WIDTH;
```

Comment:
trivial: if you're ever editing this file again, its way more common to put `const` first

Reply:
At least in the line above, moving `const` to the front would change the semantics to a non-const point to constant memory -- to avoid this distinction, I usually put the const last.

Reply:
If we were starting writing C in a vacuum, I'd agree that placing the `const` right next to the variable name is more consistent as it always means the same thing here - the variable is constant, regardless of whether it's a pointer or not. However, I agree with Tom that in the existing body of C already written, `const size_t foo` seems way more common. So, there's a conflict here, unfortunately, between your sensible argument, and existing usage.

Reply:
That's fine, I'll change it.